lib/gears/surface.lua

```diff
@@ -45,17 +45,23 @@ function surface.load_uncached_silently(_surface, default)
     if cairo.Surface:is_type_of(_surface) then
         return _surface
     end
+
+    local err, s
+
     -- Strings are assumed to be file names and get loaded
     if type(_surface) == "string" then
-        local err
         file = _surface
-        _surface, err = capi.awesome.load_image(file)
+
+        -- `s` is an undocumented workaround until #908 is implemented
+        _surface, err, s = capi.awesome.load_image(file)
+
         if not _surface then
             return get_default(default), err
         end
     end
+
     -- Everything else gets forced into a surface
-    return cairo.Surface(_surface, true)
+    return s or cairo.Surface(_surface, true)
```

Comment:
Also, why does the commit message refer to #908? That seems quite unrelated.

Reply:
https://github.com/awesomeWM/awesome/issues/1235

Reply:
I don't get it. O.O

Reply:
Long version:

This hack is required because we have something called `capi.awesome.load_image()` that should not exist and should have used LGI all along, but could not. Now it could (https://github.com/awesomeWM/awesome/issues/908#issuecomment-262906787) but that would (maybe?) cause us to bump the minimum LGI/GDK version.

The second problem is that GDK prints a big fat warning on stderr when X11 isn't present and this causes the tests to fail on the CI. Implementing a REGEX to ignore that warning could be a solution and the `gears.surface` part of the hack could be avoided. But it would also set a precedent and pollute the already messy code that generates the images.

So this PR adds a couple of ugly commits to work around the issue. It has to do so to be able to display the titlebar **and** the layoutlist in the other PR. It is also required to "complete" the shims to be able to render arbitrary screenshots and even potentially animated SVG.